modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -1645,6 +1645,108 @@ CV__DNN_INLINE_NS_BEGIN
+          */
+         CV_WRAP YOLODetectionModel(const String& model, const String& config, const int version);
+
+         /**
+          * @brief Create model from onnx graph.
+          * @param[in] onnx path to onnx graph.
+          * @param[in] version Version of the yolo model.
+          */
+         CV_WRAP YOLODetectionModel(const String& onnx, const int version);
+
+         CV_DEPRECATED_EXTERNAL  // avoid using in C++ code (need to fix bindings first)
+         YOLODetectionModel();
+
+         /** @brief Given the @p input frame, create input blob, run net and return result detections.
+          *  @param[in]  frame  The input image.
+          *  @param[out] classIds Class indexes in result detection.
+          *  @param[out] confidences A set of corresponding confidences.
+          *  @param[out] boxes A set of bounding boxes.
+          *  @param[in] confThreshold A threshold used to filter boxes by confidences.
+          *  @param[in] nmsThreshold A threshold used in non maximum suppression.
+          */
+         CV_WRAP void detect(InputArray frame, CV_OUT std::vector<int>& classIds,
+                             CV_OUT std::vector<float>& confidences, CV_OUT std::vector<Rect2d>& boxes,
+                             float confThreshold = 0.5f, float nmsThreshold = 0.0f);
+
+        /**
+         * @brief Static method for post-processing the detections.
+         *
+         * This method applies scaling and transformation to the detected bounding boxes and
+         * filters them using non-maximum suppression and confidence thresholding.
+         *
+         * @param[in] detections Input detections from the network.
+         * @param[out] boxes Processed bounding boxes after scaling and NMS.
+         * @param[out] confidences Confidence scores for the processed boxes.
+         * @param[out] classIds Class IDs for each processed box.
+         * @param[in] confThreshold Threshold for filtering boxes by confidence.
+         * @param[in] nmsThreshold Threshold for non-maximum suppression.
+         * @param[in] yoloVersion The version of the YOLO model used.
+         * @param[in] darknet Boolean to indicate if darknet architecture is used.
+         * @param[in] nmsAcrossClasses Boolean to indicate if NMS is to be applied across classes.
+         * @param[in] frameWidth Width of the input frame.
+         * @param[in] frameHeight Height of the input frame.
+         */
+         CV_WRAP static void postProccess(
+            std::vector<Mat>& detections,
+            CV_OUT std::vector<Rect2d>& boxes,
+            CV_OUT std::vector<float>& confidences,
+            CV_OUT std::vector<int>& classIds,
+            const float confThreshold,
+            const float nmsThreshold,
+            const int yoloVersion,
+            const bool darknet,
+            const bool nmsAcrossClasses,
+            int frameWidth,
+            int frameHeight
+            );
```

Comment:
Please keep required parameters at the beginning but move optional with some default values to the end:
```cpp
std::vector<Mat>& detections,
CV_OUT std::vector<Rect2d>& boxes,
CV_OUT std::vector<float>& confidences,
CV_OUT std::vector<int>& classIds,
int frameWidth,
int frameHeight
const float confThreshold = 0.5f,
const float nmsThreshold = 0.4f,
const bool nmsAcrossClasses = true,
```

The following parameters seems redundant and can be replaced to `detections` shapes analysis.
```
const int yoloVersion,
const bool darknet,
```

Reply:
done